doc/Doxyfile

```diff
@@ -2049,7 +2049,7 @@ INCLUDE_FILE_PATTERNS  =
 # recursively expanded use the := operator instead of the = operator.
 # This tag requires that the tag ENABLE_PREPROCESSING is set to YES.
 
-PREDEFINED             = "DOXYGEN_SHOULD_SKIP_THIS=1" "DOXYGEN_DOCUMENTATION_BUILD"
+PREDEFINED             = "DOXYGEN_DOCUMENTATION_BUILD"
```

Comment:
`DOXYGEN_SHOULD_SKIP_THIS` is already in use in other parts of the code, so you cannot just remove it. I understand that its an unfortunate name because of double negatives, thats why I added `DOXYGEN_DOCUMENTATION_BUILD`. My thinking was to migrate to that later maybe even in smaller steps.

Otherwise the changes in `config.hpp` look good to me, nice catch with that extra `#endif` I somehow got in by accident. I'll close the PR I added to your repo.